readthedocs/core/resolver.py

```diff
@@ -135,7 +139,9 @@ def resolve_domain(project, private=None):
     relation = project.superprojects.first()
     subdomain = getattr(settings, 'USE_SUBDOMAIN', False)
     prod_domain = getattr(settings, 'PRODUCTION_DOMAIN')
-    public_domain = getattr(settings, 'PUBLIC_DOMAIN', prod_domain)
+    public_domain = getattr(settings, 'PUBLIC_DOMAIN', None)
+    if public_domain is None:
+        public_domain = prod_domain
```

Comment:
Yup, that's the intended effect. If the `PUBLIC_DOMAIN` setting is missing completely, then it should default to `None`. If `PUBLIC_DOMAIN` was either missing completely, or set to None for some other reason, assume `PROD_DOMAIN`. This is all to ensure `PUBLIC_DOMAIN`, as it isn't an option setting anymore.